cloud/pkg/devicecontroller/controller/upstream.go

```diff
@@ -113,6 +126,34 @@ func (uc *UpstreamController) updateDeviceStatus() {
 			return
 		case msg := <-uc.deviceStatusChan:
 			klog.Infof("Message: %s, operation is: %s, and resource is: %s", msg.GetID(), msg.GetOperation(), msg.GetResource())
+			resource := msg.GetResource()
+			if strings.Contains(resource, constants.ResourceTypeDeviceRegister) {
+				// receive device.CurrentNode msg
+				nodeID, err := messagelayer.GetDeviceID(msg.GetResource())
+				if err != nil {
+					klog.Warningf("Get nodeID failed due to error %v", err)
+					continue
+				}
+				updatedDeviceName := msg.GetContent().(string)
+				value, ok := uc.dc.deviceManager.DeployedDevice.Load(updatedDeviceName)
+				if ok {
+					updatedDevice := value.(*v1beta1.Device)
+					updatedDevice.Status.CurrentNode = nodeID
+					uc.dc.deviceManager.DeployedDevice.Store(updatedDeviceName, updatedDevice)
+				}
+			} else if strings.Contains(resource, constants.ResourceTypeDeviceMigrate) {
+				hubInfo := msg.GetContent().(commonmodel.HubInfo)
+				nodeID := hubInfo.NodeID
+				err := uc.dc.mapperDeleted(nodeID)
+				if err != nil {
+					klog.Warningf("Failed to delete mapper record about node %s", nodeID)
+				}
+				err = uc.dc.deviceMigrated(nodeID)
```

Comment:
Does the `uc.dc.deviceMigrated(nodeID)` depend on the `uc.dc.mapperDeleted(nodeID)`? If dependent, is there a problem with out handing error?